core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -1158,19 +1159,28 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
       newConsumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, testGroupId)
       newConsumerConfig.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, testClientId)
       val consumer = createConsumer(configOverrides = newConsumerConfig)
+      val latch = new CountDownLatch(1)
       try {
         // Start a consumer in a thread that will subscribe to a new group.
         val consumerThread = new Thread {
           override def run {
             consumer.subscribe(Collections.singleton(testTopicName))
-            while (true) {
-              consumer.poll(5000)
-              consumer.commitSync()
+            consumer.poll(JDuration.ZERO) // Make the subscription take effect immediately
```

Comment:
Does single call to consumer.poll(0) guarantee consumer assignment?
maybe we need to wait until [assignment.](https://github.com/apache/kafka/blob/trunk/core/src/test/scala/integration/kafka/api/LegacyAdminClientTest.scala#L150)